sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JdbcSourceProvider.scala

```diff
@@ -17,12 +17,20 @@
 
 package org.apache.spark.sql.execution.datasources.jdbc
 
+import scala.collection.JavaConverters._
+
 import org.apache.spark.sql.{AnalysisException, DataFrame, SaveMode, SQLContext}
 import org.apache.spark.sql.execution.datasources.jdbc.JdbcUtils._
 import org.apache.spark.sql.sources.{BaseRelation, CreatableRelationProvider, DataSourceRegister, RelationProvider}
+import org.apache.spark.sql.sources.v2.{DataSourceOptions, StreamingWriteSupportProvider}
+import org.apache.spark.sql.sources.v2.writer.streaming.StreamingWriteSupport
+import org.apache.spark.sql.streaming.OutputMode
+import org.apache.spark.sql.types.StructType
 
-class JdbcRelationProvider extends CreatableRelationProvider
-  with RelationProvider with DataSourceRegister {
+class JdbcSourceProvider extends CreatableRelationProvider
```

Comment:
I'm curious why to change this? Is it related change?

Reply:
As I know, RelationProvider is used for batch computation, not for streaming. So I make this change, just as `KafkaSourceProvider` does.

Reply:
I think it's OK, because the `shortName` method is the same, so users using`format("jdbc")` are not affected.

Reply:
Although it is rare case, but it's still possible that if you use full classname then this will break it. Anyway, I don't see strong reason to support this renaming. You can see others' opinions too.